packages/frontend/src/components/profile/remove_account/RemoveAccount.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { FC, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Translate } from 'react-localize-redux';
+import { useSelector, useDispatch } from 'react-redux';
+
+import TrashIcon from '../../../images/icon-trash';
+import { switchAccount } from '../../../redux/actions/account';
+import { selectAccountId } from '../../../redux/slices/account';
+import { isEncrypted } from '../../../utils/encryption/keys';
+import { wallet } from '../../../utils/wallet';
+import EnterPasswordForm from '../../accounts/EnterPasswordForm';
+import FormButton from '../../common/FormButton';
+import Modal from '../../common/modal/Modal';
+import { StyledRemoveAccount } from './ui';
+import RemoveAccountForm from './ui/RemoveAccountForm';
+
+const modalStyles = {
+    maxWidth: '486px',
+};
+
+const buttonStyle = {
+    marginTop: 0,
+};
+
+const RemoveAccount: FC = () => {
+    const dispatch = useDispatch();
+    const { t } = useTranslation();
+    const accountId = useSelector(selectAccountId);
+    const [showPasswordConfirmation, setShowPasswordConfirmation] = useState(false);
+    const [showRemoveAccountConfirmation, setShowRemoveAccountConfirmation] =
+        useState(false);
+    const [isModalOpen, setIsModalOpen] = useState(
+        showPasswordConfirmation || showRemoveAccountConfirmation
+    );
+
+    useEffect(
+        () => setIsModalOpen(showPasswordConfirmation || showRemoveAccountConfirmation),
+        [showPasswordConfirmation, showRemoveAccountConfirmation]
+    );
+
+    const openPasswordConfirmation = () => setShowPasswordConfirmation(true);
+    const closePasswordConfirmation = () => setShowPasswordConfirmation(false);
+
+    const openDeletionConfirmation = () => setShowRemoveAccountConfirmation(true);
+    const closeDeletionConfirmation = () => {
+        closePasswordConfirmation();
+        setShowRemoveAccountConfirmation(false);
+    };
+
+    const startConfirmation = () => {
+        if (isEncrypted()) {
+            openPasswordConfirmation();
+        } else {
+            openDeletionConfirmation();
+        }
+    };
+
+    const onPasswordConfirmation = () => {
+        openDeletionConfirmation();
+        closePasswordConfirmation();
+    };
+
+    const processAccountDeletion = async () => {
+        const walletAccounts = await wallet.removeWalletAccount(accountId);
```

Comment:
let's move logic to ./lib from view, [like here](https://github.com/mynearwallet/near-wallet/pull/79/files#diff-d87c52dede03cb01cac01a4d435f85dc0d290555bf6b12bb5a9970a33bd551f0R22)